pkg/workload/sql_runner.go

```diff
@@ -47,22 +47,36 @@ type SQLRunner struct {
 
 	// The fields below are set by Init.
 	initialized bool
-	method      method
+	method      Method
 	mcp         *MultiConnPool
 }
 
-type method int
+// Method represents the protocol method used by the sql runner.
+// This is used mainly to enable / disabled prepared statements.
+type Method int
 
 const (
-	prepare method = iota
-	noprepare
-	simple
+	// Prepare: we prepare the query once during Init, then we reuse it for
+	// each execution. This results in a Bind and Execute on the server each time
+	// we run a query (on the given connection). Note that it's important to
+	// prepare on separate connections if there are many parallel workers; this
+	// avoids lock contention in the sql.Rows objects they produce. See #30811.
+	Prepare Method = iota
+	// NoPrepare: each query is issued separately (on the given connection).
+	// This results in Parse, Bind, Execute on the server each time we run a
```

Comment:
i'm pretty sure the driver will automatically prepare it and reuse the stmt, even if we don't use it by name. anyway, no need to change this now, that's beyond the scope of this PR.

although hmm maybe we should just get rid of noprepare entirely. it doesn't seem helpful really.